src/include/test/unit_api_vamana_index.cc

```diff
@@ -901,3 +902,58 @@ TEST_CASE("write and load index with timestamps", "[api_vamana_index]") {
         std::vector<uint64_t>{100}.begin()));
   }
 }
+TEST_CASE("vamana cosine distance", "[api_vamana_index]") {
```

Comment:
(Minor) Could you also add something to this test case where we open up a new index by URI and verify that we still use Cosine, and that the scores are correct? That will test the `read_types()` changes.